runtime/src/serde_snapshot.rs

```diff
@@ -321,19 +390,54 @@ where
     let mut accounts_db = AccountsDb::new_with_config(
         account_paths.to_vec(),
         cluster_type,
-        account_indexes,
+        account_secondary_indexes,
         caching_enabled,
         shrink_ratio,
     );
-    let AccountsDbFields(storage, version, slot, bank_hash_info) = accounts_db_fields;
+    let AccountsDbFields(
+        full_snapshot_storage,
+        full_snapshot_version,
+        full_snapshot_slot,
+        full_snapshot_bank_hash_info,
+    ) = snapshot_accounts_db_fields.full_snapshot_accounts_db_fields;
+
+    let incremental_snapshot_version = snapshot_accounts_db_fields
+        .incremental_snapshot_accounts_db_fields
+        .as_ref()
+        .map(|fields| fields.1);
+    let incremental_snapshot_slot = snapshot_accounts_db_fields
+        .incremental_snapshot_accounts_db_fields
+        .as_ref()
+        .map(|fields| fields.2);
+    let incremental_snapshot_bank_hash_info = snapshot_accounts_db_fields
+        .incremental_snapshot_accounts_db_fields
+        .as_ref()
+        .map(|fields| fields.3.clone());
```

Comment:
How about 

```
let (snapshot_version, snapshot_slot, snapshot_bank_hash_info) = snapshot_accounts_db_fields
        .incremental_snapshot_accounts_db_fields
        .as_ref()
        .map(
            |AccountsDbFields(
                _,
                incremental_snapshot_version,
                incremental_snapshot_slot,
                incremental_snapshot_bank_hash_info,
            )| {
                (
                    *incremental_snapshot_version,
                    *incremental_snapshot_slot,
                    incremental_snapshot_bank_hash_info.clone(),
                )
            },
        )
        .unwrap_or((
            full_snapshot_version,
            full_snapshot_slot,
            full_snapshot_bank_hash_info,
        ));
```

and removing the `unwrap_or` for each of the references below

Reply:
Love it! Done.